src/compiler/checker.ts

```diff
@@ -349,7 +349,8 @@ module ts {
                             }
                             result = undefined;
                         }
-                        else if (location.kind === SyntaxKind.SourceFile) {
+                        else if (location.kind === SyntaxKind.SourceFile ||
+                            (location.kind === SyntaxKind.ModuleDeclaration && (<ModuleDeclaration>location).name.kind === SyntaxKind.StringLiteral)) {

```

Comment:
Would `hasExternalModuleSymbol` help you out at all?

I notice we do this `name.kind === SyntaxKind.StringLiteral` thing a lot. Can we abstract this to a helper function?